src/contract/aci.ts

```diff
@@ -0,0 +1,495 @@
+  }
+
+  if (contractAddress != null) {
+    const contract = await this.getContract(contractAddress)
+    if (contract.active == null) throw new InactiveContractError(contractAddress)
+  }
+
+  const instance: ContractInstance = {
+    _aci,
+    _name: _aci.encodedAci.contract.name,
+    calldata: new Calldata([_aci.encodedAci, ..._aci.externalEncodedAci]),
+    source,
+    bytecode,
+    deployInfo: { address: contractAddress },
+    options: {
+      ...this.Ae.defaults,
+      callStatic: false,
+      fileSystem,
+      ...otherOptions
+    },
+    compilerVersion: this.compilerVersion,
+    compile: async function (_options?: {}): Promise<any> {},
+    _estimateGas: async function (_name: string, _params: any[], _options: any): Promise<any> {},
+    deploy: async function (_params: any[], _options: any): Promise<any> {},
+    call: async function (_fn: string, _params?: any[], _options?: {}): Promise<any> {},
+    decodeEvents: function (_events: Event[], { omitUnknown, ...opt }: {
+      omitUnknown?: boolean}): any {},
+    methods: undefined
+  }
+
+  if (validateBytecode != null) {
+    if (contractAddress == null) throw new MissingContractAddressError('Can\'t validate bytecode without contract address')
+    const onChanBytecode = (await this.getContractByteCode(contractAddress)).bytecode
+    const isValid: boolean = source != null
+      ? await this.compilerApi.validateByteCode(
+        { bytecode: onChanBytecode, source, options: instance.options }
+      ).then(() => true, () => false)
+      : bytecode === onChanBytecode
+    if (!isValid) throw new BytecodeMismatchError(source != null ? 'source' : 'bytecode')
+  }
+
+  /**
+  * Compile contract
+  * @alias module:@aeternity/aepp-sdk/es/contract/aci
+  * @rtype () => ContractInstance: Object
+  * @return bytecode
+  */
+  instance.compile = async (options = {}): Promise<string> => {
+    if (instance.bytecode != null) throw new IllegalArgumentError('Contract already compiled')
+    if (instance.source == null) throw new IllegalArgumentError('Can\'t compile without source code')
+    const { bytecode } = await this.compilerApi.compileContract({
+      code: instance.source, options: { ...instance.options, ...options }
+    })
+    instance.bytecode = bytecode
+    return instance.bytecode ?? ''
+  }
+
+  const sendAndProcess = async (tx: string, options: any): Promise<{
+    result?: ContractInstance['deployInfo']['result']
+    hash: string
+    tx: object
+    txData: TxData
+    rawTx: string
+  }> => {
+    const txData = await this.send(tx, options)
+    const result = {
+      hash: txData.hash, tx: TxObject({ tx: txData.rawTx }), txData, rawTx: txData.rawTx
+    }
+    if (txData.blockHeight == null) return result
+    const { callInfo } = await this.api.getTransactionInfoByHash(txData.hash)
+    Object.assign(result.txData, callInfo) // TODO: don't duplicate data in result
+    await handleCallError(callInfo, tx)
```

Comment:
```suggestion
    handleCallError(callInfo, tx)
```
handleCallError isn't async